src/renderer/views/Search/Search.js

```diff
@@ -90,26 +91,33 @@ export default defineComponent({
   },
   methods: {
     checkSearchCache: function (payload) {
-      const sameSearch = this.sessionSearchHistory.filter((search) => {
-        return search.query === payload.query && searchFiltersMatch(payload.searchSettings, search.searchSettings)
-      })
+      const searchCharLimit = this.$store.getters.getSearchCharacterLimit
 
-      if (sameSearch.length > 0) {
-        // No loading effect needed here, only rendered result update
-        this.replaceShownResults(sameSearch[0])
-      } else {
-        // Show loading effect coz there will be network request(s)
-        this.isLoading = true
-        this.searchSettings = payload.searchSettings
-
-        switch (this.backendPreference) {
-          case 'local':
-            this.performSearchLocal(payload)
-            break
-          case 'invidious':
-            this.performSearchInvidious(payload, { resetSearchPage: true })
-            break
+      if (payload.query.length <= searchCharLimit) {
+        const sameSearch = this.sessionSearchHistory.filter((search) => {
+          return search.query === payload.query && searchFiltersMatch(payload.searchSettings, search.searchSettings)
+        })
+
+        if (sameSearch.length > 0) {
+          // No loading effect needed here, only rendered result update
+          this.replaceShownResults(sameSearch[0])
+        } else {
+          // Show loading effect coz there will be network request(s)
+          this.isLoading = true
+          this.searchSettings = payload.searchSettings
+
+          switch (this.backendPreference) {
+            case 'local':
+              this.performSearchLocal(payload)
+              break
+            case 'invidious':
+              this.performSearchInvidious(payload, { resetSearchPage: true })
+              break
+          }
         }
+      } else {
+        console.warn('Search character limit is: ', searchCharLimit)
+        showToast(i18n.t('Search character limit', { searchCharacterLimit: searchCharLimit }))
```

Comment:
This few lines should be before the main code and perform an early return
No point making the remaining existing code nested when it can be avoided
```js
if (payload.query.length > searchCharLimit) {
  console.warn('Search character limit is: ', searchCharLimit)
  showToast(i18n.t('Search character limit', { searchCharacterLimit: searchCharLimit }))
  return
}
```

Reply:
I've made the changes requested